subprojects/normalization-java/src/main/java/org/gradle/api/internal/changedetection/state/LineEndingAwareResourceHasher.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.changedetection.state;
+
+import org.gradle.internal.fingerprint.LineEndingSensitivity;
+import org.gradle.internal.fingerprint.hashing.RegularFileSnapshotContext;
+import org.gradle.internal.fingerprint.hashing.ResourceHasher;
+import org.gradle.internal.fingerprint.hashing.ZipEntryContext;
+import org.gradle.internal.hash.HashCode;
+import org.gradle.internal.hash.Hasher;
+import org.gradle.internal.io.IoSupplier;
+
+import javax.annotation.Nullable;
+import java.io.File;
+import java.io.IOException;
+import java.util.Optional;
+
+/**
+ * A {@link ResourceHasher} that ignores line endings while hashing the file.  It detects whether a file is text or binary and only
+ * normalizes line endings for text files.  If a file is detected to be binary, we fall back to the existing non-normalized hash.
+ */
+public class LineEndingAwareResourceHasher implements ResourceHasher {
```

Comment:
WDYT about renaming all `LineEndingAware*` stuff to `LineEndingNormalizing*`? I think that'd better capture what these things do.

Reply:
No objections.  Done.